prototypes/ItemRepeaterPrototypes/ItemsRepeaterExperiments/AttachedBehaviors/SelectionModelPage.xaml.cs

```diff
@@ -0,0 +1,134 @@
+                selectionModel.SelectedItems.Contains(teams[TeamPresenter.GetElementIndex(control)]));
+        }
+
+        private void Control_PointerReleased(object sender, PointerRoutedEventArgs e)
+        {
+            var elementIndex = TeamPresenter.GetElementIndex(sender as UIElement);
+            if (selectionModel.SingleSelect)
+            {
+                if (e.KeyModifiers == Windows.System.VirtualKeyModifiers.Control)
+                {
+                    selectionModel.Deselect(elementIndex);
+                }
+                else
+                {
+                    selectionModel.Select(elementIndex);
+                }
+            }
+            else
+            {
+                switch (e.KeyModifiers)
+                {
+                    case Windows.System.VirtualKeyModifiers.Shift:
+                        if (PointerBehaviors.LastFocused != null)
+                        {
+                            var lastFocused = TeamPresenter.GetElementIndex(PointerBehaviors.LastFocused as UIElement);
+                            selectionModel.SetAnchorIndex(lastFocused);
+                        }
+                        selectionModel.SelectRangeFromAnchor(elementIndex);
+                        break;
+                    case Windows.System.VirtualKeyModifiers.Control:
+                        selectionModel.Deselect(elementIndex);
+                        break;
+                    default:
+                        selectionModel.Select(elementIndex);
+                        break;
+                }
+            }
+            PointerBehaviors.LastFocused = sender as UIElement;
+        }
+
+        /// <summary>
+        /// This event gets raised when an item gets selected
+        /// @WINUI This event definitely needs a list of selected and deselected items
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        private void SelectionModel_SelectionChanged(SelectionModel sender, SelectionModelSelectionChangedEventArgs args)
+        {
+            // Having the deselected items would really be handy here ...
+            foreach (var item in teams)
+            {
+                var control = TeamPresenter.TryGetElement(teams.IndexOf(item as Team)) as DependencyObject;
+                // Container not realized ... , so skip
+                if (control == null)
+                {
+                    //Debug.WriteLine("Not found: " + item.Name);
+                    continue;
+                }
+                if (selectionModel.SelectedItems.Contains(item))
+                {
+                    SelectionBehavior.SetIsSelected(control, true);
+                }
+                else
+                {
+                    SelectionBehavior.SetIsSelected(control, false);
+                }
+            }
+        }
+
+        private void SingleSelectCheckbox_Checked(object sender, RoutedEventArgs e)
```

Comment:
nit: this can be done by binding the checkbox IsChecked property to SingleSelect property of selection model.

Reply:
Since we clear selection and IsChecked is a nullable bool which would require a converter, this would be the easiest way to handle the switch.

Reply:
That is unfortunate that this needs a converter. Why clearing the selection though ? I think when going from multi to single, selection model will automatically just keep the first selected index as the single selection.

Reply:
Also, can you see if x:Bind works - that might do the conversion for you.

Reply:
Interestingly, when we switch from SingleSelect to non SingleSelect and vice versa, selection changed does not get raised, so we need to manually call the code for updating our selections.

x:Bind also requires a converter for nullable bool to bool. Since we need the extra step of updating the selections manually at some bind, I think binding won't be an option unfortunately.

Reply:
@chingucoding You can use a cast instead of a converter to bind a Boolean? property to a Boolean property:
```xml
<SelectionModel SingleSelect="{x:Bind (x:Boolean)SingleSelectCheckbox.IsChecked, Mode=OneWay}" />
```

Reply:
Didn't know that. None the less we need to perform some actions upon check/uncheck :/

Reply:
> Interestingly, when we switch from SingleSelect to non SingleSelect and vice versa, selection changed does not get raised

This sounds like a bug to me. I would expect SelectionChanged to be raised in case of changing SelectionMode from *Multiple* to *Single* while more than one item was selected at that point. Looking at the SelectionModel code [in question](https://github.com/microsoft/microsoft-ui-xaml/blob/master/dev/Repeater/SelectionModel.cpp#L57) one code comment seems to suggest the SelectionChanged event will indeed be raised. However, this is not the case. In fact, checking the code, when `SelectedIndex(firstSelectionIndexPath)` is called with the apparent intention to raise the event, the item at the firstSelectionIndexPath is already selected thanks to the code called prior to SelectedIndex(...), meaning that SelectedIndex will in fact **not** raise the event.

If this is indeed unintended behavior, I can open an issue and fix this.

Reply:
I think the issue may be caused by the fix for this issue: #1642 

Though I have to admit, I am not entirely sure when the fix was released in a nuget package, so maybe the test app did not get that fix yet.

Reply:
The fix for #1642 is in the current pre-release build so if you see that behavior there then it is likely a bug.. I would expect SelectionChanged to be raised in that case since selection has indeed changed.

Reply:
@chingucoding PR https://github.com/microsoft/microsoft-ui-xaml/pull/2359 is now included in the new prerelease (Microsoft.UI.Xaml v2.4.0-prerelease.200506001) so SelectionChanged should now be raised if required when switching from *Multiple* to *Single*.

I see you are also clearing the selection when going from selection mode *Single* to *Multiple*. Why is that? Do you want to "start fresh" when the selection mode is changed?

Reply:
The problem was, that without the SelectionChanged event we are in a dirty state when we switch selection modes, to prevent that, we clear selection.

Reply:
But is switching from Single to Multiple selection mode supposed to change the selection and raise a SelectionChanged event? I would expect the selection to remain intact.

Reply:
That's a good point. Thanks to your fix, that logic is gone now anyway.